### ISSUE TITLE
Potential bug when normalizing camera

### ISSUE THREAD

**Issue by cwchenwang** [hwjiang1510/Real3D#8]
When building canonical pose, the rotation is not identity, is this a bug or my understanding is wrong?
```
    def _build_canonical_pose(self, scale):
        '''
        build canonical pose in opencv pose convension
        the canonical pose has an identity rotation and a translation along the z-axis
        '''
        canonical_pose = torch.tensor([[0.0, 0.0, 1.0, scale],
                                        [1.0, 0.0, 0.0, 0.0],
                                        [0.0, 1.0, 0.0, 0.0],
                                        [0.0, 0.0, 0.0, 1.0]]) @ opengl_to_cv2
        return canonical_pose
```

**Comment by hwjiang1510** [comment]
Hi,

Thanks for your interest in our work!

This  pose does have a identity rotation, but based on the coordinate frame definition of the model and the dataset, we need to permute the axis. With the canonical pose, the rendered image should be in the same viewpoint of the input.

**Comment by cwchenwang** [comment]
Hi, thank you a lot for your quick reply. Could you explain more what coordinate frame of the model is? To my understanding, the function should build the canonical pose in OpenCV (and the dataest is also in OpenCV), so I don't understand why we should permute like this and also use opengl_to_cv2 conversion.

**Comment by hwjiang1510** [comment]
Hi,

Yes, your understanding is generally correct. 

While we call the model's coordinate system as "opencv-style", it actually permutes the x-y-z axis to z-x-y axis. We still call it as "opencv-style" in the comments of the code as it doesn't inverse the axis, contrasting "opengl-style".

This design originates from LRM (see their section 4.2), and the open-sourced OpenLRM and TripoSR both follow this setting but with minor differences. For example, OpenLRM's canonical pose is defined [here](https://github.com/3DTopia/OpenLRM/blob/3108b6902d27912e1dcdce76970eab51436f5b21/openlrm/runners/infer/lrm.py#L131).

We use the checkpoint of TripoSR for fine-tuning, so Real3D inherents the design of this coordinate frame system. If you modify the canonical pose as x-y-z permutation and use it to render the reconstruction, it will not output the image with the same viewpoint of the input.

**Comment by cwchenwang** [comment]
I think I understand now, thank you very much!
